fms_fsdp/config/training.py

```diff
@@ -20,7 +20,7 @@ class train_config:
     # fsdp policies
     mixed_precision: bool = True
     fsdp_activation_checkpointing: bool = False
-    selective_checkpointing: int = 1
+    selective_checkpointing: tuple[int, int] = (1, 1)  # ac m blocks every n blocks
```

Comment:
is there a simpler way to express this? maybe something like "percent of blocks that should be checkpointed" and we convert to the tuple format internally?

Reply:
initially something like this was in my mind (very rough, but you get the point):
```python
# given percentage x
count = 1

# looping over all block_idx
if block_idx * x >= count: 
    count += 1
    return True
```

this should roughly do it, but missing a few properties, e.g. not evenly spaced inside a subgroup.   

e.g. if percentage is 1/20,  then either the 0th or 19th will be applied (on edges). while the implementation in this PR would apply to 9th/10th block.  And this may cause some edge cases.

we can of course prettify this as well. Let me know what you think, and maybe you have better ideas on this.